core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -232,6 +240,14 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
     }
   }
 
+  /** Add Credentials to the currently logged in user. */
+  private def addCredentials(creds: Credentials): Unit = {
+    logInfo(s"Adding ${creds.numberOfTokens()} tokens and ${creds.numberOfSecretKeys()} secret" +
+      s"keys to the current user's credentials.")
+
+    UserGroupInformation.getCurrentUser().addCredentials(creds)
```

Comment:
This looks like `SparkHadoopUtil.addCurrentUserCredentials`. That's only implemented on `YarnSparkHadoopUtil` for historical reasons, but since we dropped Hadoop 1.x support, the implementation can move to `core/` now, and you'd avoid this copy of that code.